### PR TITLE
[core] Fix #3427 - Stop printing usage text on error

### LINKED ISSUE #3427: [core] Stop printing CLI usage text when exiting due to invalid parameters
**Is your feature request related to a problem? Please describe.**
When making a mistake in CLI parameters a wall of text is printed and the actual error message is completely drowned out. Also this is a real problem with the `-help` option. Because of #3424, `pmd --help` is actually incorrect (`pmd -help` should be used instead). But since the usage text is printed, we have the impression that it actually worked, only to be disappointed that this wall of text doesn't mention the CLI syntax. It comes off as buggy and unhelpful.

**Describe the solution you'd like** Just print the error message, followed by the line `Run with --help to get more information` or something like that. Note that `--help` doesn't exist yet (#3424)

### PR DESCRIPTION
**Pull request by oowekyala**
## Related issues

- Fixes #3427, also for CPD

## Ready?

- [x] Added unit tests for fixed bug/feature
- [ ] Passing all unit tests
- [ ] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [ ] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">This changeset changes 0 violations,<br />
introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/9da201e8b83e41d2a622422442d4a745/diff/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-core/src/main/java/net/sourceforge/pmd/cli/internal/CliMessages.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+public final class CliMessages {
+
+    private CliMessages() {
+        // utility class
+    }
+
+    public static String errorDetectedMessage(int errors, String program) {
+        String anError = errors == 1 ? "An error" : errors + " errors";
+        return anError + " occurred while executing " + program + ".\n"
+            + "Run with --debug to see a stack-trace.\n"
```
Comment (adangel) [inline pmd-core/src/main/java/net/sourceforge/pmd/cli/internal/CliMessages.java:19]: I think, CPD doesn't have a `--debug` flag... at least we don't have it documented: https://pmd.github.io/latest/pmd_userdocs_cpd.html
I'll create a new issue, this should be sorted at some point...

**Review by adangel — APPROVED** [review]
Thanks!

File: pmd-core/src/main/java/net/sourceforge/pmd/cli/internal/CliMessages.java (hunk 1 above)
Comment (adangel) [reply pmd-core/src/main/java/net/sourceforge/pmd/cli/internal/CliMessages.java:19]: See #3796

**Pull request merged**